samples/net/sockets/http_server/src/server.c

```diff
@@ -0,0 +1,889 @@
+		printf("HTTP1_REQUEST.\n");
+		ctx_client->server_state = HTTP1_REQUEST;
+	} else {
+		ctx_client->is_http2 = 1;
+		ctx_client->offset -= strlen(preface);
+		memmove(ctx_client->buffer,
+			ctx_client->buffer + strlen(preface),
+			ctx_client->offset);
+
+		if (ctx_client->offset == 0) {
+			ctx_client->server_state = CLOSING;
+		} else {
+			ctx_client->server_state =
+				determine_server_state(ctx_client->buffer);
+		}
+	}
+
+	return 0;
+}
+
+int handle_http1_request(struct http2_server_ctx *ctx_server,
+	struct http2_client_ctx *ctx_client, int client_index)
+{
+	const char *data;
+	int len;
+	int total_received = 0;
+	int offset = 0;
+
+	http_parser_init(&parser, HTTP_REQUEST);
+	http_parser_settings_init(&parserSettings);
+	parserSettings.on_header_field = on_header_field;
+	http_parser_execute(&parser, &parserSettings,
+		ctx_client->buffer + offset, ctx_client->offset);
+
+	total_received += ctx_client->offset;
+	offset += ctx_client->offset;
+
+	if (offset >= MAX_HTTP_RESPONSE_SIZE)
+		offset = 0;
+
+	if (has_upgrade_header == false) {
+		const char *response = "HTTP/1.1 101 Switching Protocols\r\n"
+				       "Connection: Upgrade\r\n"
+				       "Upgrade: h2c\r\n"
+				       "\r\n";
+		if (sendall(ctx_client->client_fd, response, strlen(response)) <
+			0)
+			close_client_connection(ctx_server, client_index);
+
+		memset(ctx_client->buffer, 0, sizeof(ctx_client->buffer));
+		ctx_client->offset = 0;
+		ctx_client->server_state = CLOSING;
+
+	} else {
+		data = content;
+		len = sizeof(content);
+
+		http_parser_init(&parser, HTTP_REQUEST);
+		http_parser_settings_init(&parserSettings);
+		parserSettings.on_url = on_url;
+		http_parser_execute(&parser, &parserSettings,
+			ctx_client->buffer, sizeof(ctx_client->buffer));
+
+		if (strcmp(url_buffer, "/") == 0) {
+			sprintf(http_response,
+				"HTTP/1.1 200 OK\r\n"
+				"Content-Type: text/html\r\n"
+				"Content-Encoding: gzip\r\n"
+				"Content-Length: %d\r\n\r\n",
+				len);
+			if (sendall(ctx_client->client_fd, http_response,
+				    strlen(http_response)) < 0) {
+				close_client_connection(
+					ctx_server, client_index);
+			} else {
+
+				if (sendall(ctx_client->client_fd, data, len) <
+					0) {
+					LOG_ERR("sendall failed");
+					close_client_connection(
+						ctx_server, client_index);
+				}
+			}
+		} else {
+			const char *not_found_response =
+				"HTTP/1.1 404 Not Found\r\n"
+				"Content-Length: 9\r\n\r\n"
+				"Not Found";
+			if (sendall(ctx_client->client_fd, not_found_response,
+				    strlen(not_found_response)) < 0) {
+				close_client_connection(
+					ctx_server, client_index);
+			}
+		}
+		close_client_connection(ctx_server, client_index);
+		memset(ctx_client->buffer, 0, sizeof(ctx_client->buffer));
+		ctx_client->offset = 0;
+		ctx_client->server_state = CLOSING;
+	}
+
+	return 0;
+}
+
+int handle_initial_state(struct http2_client_ctx *ctx_client)
+{
+	if (!ctx_client->is_http2) {
```

Comment:
There shouldn't be a need for `is_http2` variable - whether we deal with HTTP1 or HTTP2 request should be evaluated based on the current state we're in.